### PR TITLE
Remove network_status_tracker code

### PR DESCRIPTION
**Pull request by yang-g**
This would help when the server have a lot of short lived incoming connections. It is not sufficient, but is an improve from the current status.

### CODE REVIEW

File: src/core/lib/iomgr/network_status_tracker.c [hunk 1]
```diff
@@ -40,58 +40,76 @@ typedef struct endpoint_ll_node {
...
 void grpc_network_status_register_endpoint(grpc_endpoint *ep) {
-  gpr_mu_lock(&g_endpoint_mutex);
-  if (head == NULL) {
-    head = (endpoint_ll_node *)gpr_malloc(sizeof(endpoint_ll_node));
-    head->ep = ep;
-    head->next = NULL;
+  shard_type *s = &g_shards[shard_idx(ep)];
+  gpr_mu_lock(&s->mu);
+  if (s->head == NULL) {
+    s->head = (endpoint_ll_node *)gpr_malloc(sizeof(endpoint_ll_node));
```
Comment (dgquintas) [inline src/core/lib/iomgr/network_status_tracker.c]: you could do the malloc outside the lock onto a temp and then copy it to s->head while holding the lock.

File: src/core/lib/iomgr/network_status_tracker.c [hunk 2]
```diff
@@ -40,58 +40,76 @@ typedef struct endpoint_ll_node {
...
+    s->head->ep = ep;
+    s->head->next = NULL;
   } else {
-    endpoint_ll_node *prev_head = head;
-    head = (endpoint_ll_node *)gpr_malloc(sizeof(endpoint_ll_node));
-    head->ep = ep;
-    head->next = prev_head;
+    endpoint_ll_node *prev_head = s->head;
+    s->head = (endpoint_ll_node *)gpr_malloc(sizeof(endpoint_ll_node));
```
Comment (dgquintas) [inline src/core/lib/iomgr/network_status_tracker.c]: idem

File: src/core/lib/iomgr/network_status_tracker.c [hunk 3]
```diff
@@ -40,58 +40,72 @@ typedef struct endpoint_ll_node {
...
-    gpr_free(curr);
-    curr = next;
+  for (size_t i = 0; i < SHARD_COUNT; i++) {
+    shard_type *s = &g_shards[i];
+    gpr_mu_destroy(&s->mu);
   }
-  gpr_mu_destroy(&g_endpoint_mutex);
+}
+
+// pointer hash function copied from timer_generic.c
```
Comment (ctiller) [inline src/core/lib/iomgr/network_status_tracker.c]: Can I (beg perhaps) that this be moved to useful.h... it's getting duplicated around a bit too much.

```c
#define GPR_HASH_POINTER(x, range) (((x) >> 4) ^ ((x) >> 9) ^ ((x) >> 14)) % (range)
```

Comment (yang-g) [inline src/core/lib/iomgr/network_status_tracker.c]: Done. 3 uses that I can find have been updated to use the useful.h macro. PTAL

**Comment by makdharma** [comment]
The code in network_status_tracker.c is currently not used. We only implemented one half of the feature here. We currently depend upon objc code to handle network events. In the short term it may be easier to simply return from the function calls without doing any work. @yang-g lmk if  you need me to submit another PR to do that.

**Comment by yang-g** [comment]
Thanks for clarifying. I will update this PR so that the second commit is preserved.

**Comment by yang-g** [comment]
Test flake issue: #9668

**Pull request merged**